File: examples/archive_userspace_programs/userspace.py
import pruss
from pruss import *

pss = PRUSS.get()
p1 = pss.pru1
p1.enable()

s = input("Enter a string to send to pruss_api_pru1 \n")
p1.sendMsg(s)

print("Message received: ")
print (p1.getMsg())
p1.disable()
pss.shutDown()

File: examples/firmware_examples/example2-rpmsg-pru1/userspace.cpp
#include <iostream>
#include "pruss.h"

using namespace std;

int main()
{
	PRUSS& p = PRUSS::get();
	PRU p1 = p.pru1;
	p1.enable();

	string s;
	cout << "Enter a message        : ";
	getline(cin, s);
	p1.sendMsg_string(s);

	cout << "Message from PRU       : "<< p1.getMsg();
	p1.disable();
	p.shutDown();

}
